### PR TITLE
Executor: Add readme.md describing executors

### PR DESCRIPTION
**Pull request by sven-oly**
This is a start on documentation of executors function, inputs, and outputs.

### CODE REVIEW

File: executors/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+## Requirements
+An executor must run as a command line program, accepting data from the standard input stream (stdin). The protocol expected is described below.
+
+Each executor should accept JSON data describing the test to be performed from stdin. The executor may run a single test on each execution or may perform multipe tests until it receives a command to stop its operation.
+
+Tests should be stateless, i.e., each test should create a test environment that does not depend on the results of previous tests or stored states.
+
+## Protocol for communication with testDriver
+
+The testDriver program initiates an executor a command line string such as "nodejs executor.js". TestDriver then sends information to the stdin of the running executor program. These instructions are defined below.
```
Comment (gnrunge) [inline executors/README.md:21]: ... executor a command line string ... ---> executor on command line ...

File: executors/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+Empty lines should be ignored by the executor.
+
+Output from an executor is strictly given through its STDOUT, not via files. Returned values should be in JSON format.
+
+### Errors and exceptions
+Exceptions and errors may sometimes occur in the execution. An executor may return errors in two ways:
+* The executor should return JSON data indicating all relevant parameters such as "label".
+
+  * A JSON key "error" should be included, containing a string describing the error condiont.
```
Comment (gnrunge) [inline executors/README.md:38]: Indent wrong.

Comment (gnrunge) [inline executors/README.md:38]: error condiont ---> error condition (no?)

File: executors/README.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+Executors are provided for NodeJS and Rust that actually call I18N functions in the respective platforms. In addition, a simple Python program example is an example of how a user may create and run additional executors in the DDT framework under the testDriver.
+
+Executors accept and parse data from stdin, implementing the protocol described above.
+
+* NodeJS: The file executor.js creates objects from classes of the [ECMAScript Intl object]https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl). Routines implement the following types of tests
+  * collation: Pairs of strings are compared using the default locale. A numeric result of -1, 0, or 1 is returned in the JSON item "result".
+  * number formatting:  The "options" key indicates the parameters applied to
+  * display names: Accepts locale of output
+    * Includes "calendar", "currency", "dateTimeField", "language", "region", and "script.
+    * An input value. The input depends on the type and may be the code for a language, e.g., "fr-CA", the code for a region or country, e.g., "NO", etc.
```
Comment (gnrunge) [inline executors/README.md:56]: Hmm, fr-CA contains but is more than a language code since it contains a region as well.

**Review by gnrunge — COMMENTED** [review]
Editorial comment: most lines are very long while others are limited to 80 characters/line.

File: executors/README.md (hunk 1 above)
Comment (sven-oly) [reply executors/README.md:21]: Thx.

File: executors/README.md (hunk 2 above)
Comment (sven-oly) [reply executors/README.md:38]: Intended.

**Review by gnrunge — APPROVED** [review]

**Pull request merged**